packages/boilerplate-generator/generator.js

```diff
@@ -23,13 +26,30 @@ export class Boilerplate {
   // purpose is to allow you to specify data that you might not know at
   // the time that you construct the Boilerplate object. (e.g. it is used
   // by 'webapp' to specify data that is only known at request-time).
+  // this returns a stream
   toHTML(extraData) {
-    if (!this.baseData || !this.template) {
+    if (!this.baseData || !this.headTemplate || !this.closeTemplate) {
       throw new Error('Boilerplate did not instantiate correctly.');
     }
 
-    return  "<!DOCTYPE html>\n" +
-      this.template({ ...this.baseData, ...extraData });
+    const data = {...this.baseData, ...extraData};
+    const start = "<!DOCTYPE html>\n" + this.headTemplate(data);
+
+    const { body, dynamicBody } = data;
+
+    const end = this.closeTemplate(data);
+    const response = stream.create();
+    
+    response.append(Buffer.from(start))
```

Comment:
Let's be explicit about passing `"utf8"` as the encoding type as the second argument to `Buffer.from`.